<a href="https://colab.research.google.com/github/vncspelegrini/execiciosPython/blob/main/Python_3_Conceitos_e_Aplica%C3%A7%C3%B5es_Uma_abordagem_did%C3%A1tica_S%C3%A9rgio_Luiz_Banin.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Python 3 - Conceitos e Aplicações - Uma abordagem didática - Sérgio Luiz Banin

https://integrada.minhabiblioteca.com.br/reader/books/9788536530253/pageid/0

pag 47. 1. Faça X = 0.0 e Y = 18. Verifique o tipo de dado que o Python atribuiu a cada um. Faça Z = X + Y e verifique o resultado calculado e armazenado em Z. Verifique com qual tipo de dado foi criado o objeto Z.

In [ ]:
X = 0.0
Y = 18
print(type(X))
print(type(Y))
Z = X + Y
print(Z)
print(type(Z))

<class 'float'>
<class 'int'>
18.0
<class 'float'>


pag 47 2. Atribua um valor qualquer a um objeto a (minúsculo). Utilize o comando type ou o comando print com o objeto A (maiúsculo). Relate o que aconteceu.

In [ ]:
a = 1
print(type(A))


NameError: name 'A' is not defined

pag 47. 3. No IDLE, faça A = “Questão 3”, B = 25 e C = 3.9. Utilize o comando type para verificar qual é o tipo de dado dos objetos A, B e C.

In [ ]:
A = 'Questão 3'
B = 25
C = 3.9
print(type(A))
print(type(B))
print(type(C))

<class 'str'>
<class 'int'>
<class 'float'>


pag. 47 - 4. Reproduza em um programa todos os casos de operações aritméticas do Quadro 2.1, para A = 14 e B = 5, e compare os valores obtidos por você com os valores esperados constantes do quadro.

In [ ]:
A = 14
B = 5
print(A + B)
print(A - B)
print(A * B)
print(A / B)
print(A // B)
print(A % B)
print(A ** B)

19
9
70
2.8
2
4
537824


pag 47 - 5. Escreva a sequência de comandos necessária para o cálculo da área de um triângulo de base 9 e altura 6.


In [ ]:
Base = 9
Altura = 6
Area = (Base * Altura) / 2
print(Area)

27.0


pag 47 - 6. Refaça o exercício 5 alterando-o de modo que a base e a altura do triângulo sejam lidas do teclado. Considere-as números reais.

In [ ]:
Base = float(input('Digite a base do triângulo: '))
Altura = float(input('Digite a altura do triângulo: '))
Area = (Base * Altura) / 2
print(Area)

Digite a base do triângulo: 9
Digite a altura do triângulo: 6
27.0


pag 48 - 7. Escreva a sequência de comandos para calcular o salário bruto de um pro-fissional que ganha por hora, sabendo que ele ganha R$ 14,25/h e trabalhou 163 horas normais e 20 horas extras (pagam o dobro).

In [ ]:
Valor_Hora = 14.25
Valor_Hora_Extra = Valor_Hora * 2
Horas_Trabalhadas = 163
Horas_Trabalhadas_Extra = 20
Salario_Bruto = (Horas_Trabalhadas * Valor_Hora) + (Horas_Trabalhadas_Extra * Valor_Hora_Extra)
print(Salario_Bruto)

2892.75


8. Escreva em Python as seguintes expressões aritméticas para as fórmulas a seguir e teste as quatro primeiras para os valores A = 4, B = 5, C = 1 e a última para os valores x1
= 1, y1


In [ ]:
from math import sqrt

A = 4
B = 5
C = 1
x1 = 1
y1 = 8

R = (A + B) / 2
print(R)

x = -B + sqrt(B ** 2 - 4 * A * C) / (2 * A)
print(x)

R = 3 * A + 2 * C / A + B
print(R)

z = 7.6 * A - B ** 1.7
print(z)

d = sqrt((x1 - x) ** 2 + y1 - y2 ** 2)
print(d)

4.5
-4.625
17.5
14.974153431999758
3.8263069662534916


9. Escreva um programa que leia do teclado as coordenadas (x1 , y1) do ponto 1 e (x2, y2) do ponto 2. Utilizando a expressão do item 8.e, determine a distância
entre esses dois pontos e exiba-a na tela com três casas decimais. Teste-o com os dados da tabela a seguir.
x1
0,0 2,0
-3,0 0,0 8,2 6,9
y1
0,0 1,0 1,5 3,5 2,5 2,0
x2
3,0 0,0 7,1 0,0
-5,0 16,0
y2
4,0 5,0 5,5 7,0
-5,0 -1,8
d
5,000 4,472
10,863 3,500 15,182 9,862


In [ ]:
x1 = float(input('Digite o valor de x1: '))
y1 = float(input('Digite o valor de y1: '))
x2 = float(input('Digite o valor de x2: '))
y2 = float(input('Digite o valor de y2: '))
d = ((x2 - x1)**2 + (y2 - y1)**2)**(1/2)
print(d)

Digite o valor de x1: 0.0
Digite o valor de y1: 0.0
Digite o valor de x2: 3
Digite o valor de y2: 5
5.830951894845301


10. Um vendedor ambulante vendeu os produtos indicados na tabela a seguir. Informe quanto ele faturou com cada produto e quanto ele faturou no total.

Produto
Boneco Malandrinho
Spinner Pequeno
Cubo Mágico

Quantidade vendida
17
36
7

Valor unitário R$
18,50
12,00
5,90

Todos os dados devem ser lidos do teclado, sendo que o nome do produto é string, a quantidade vendida é um número inteiro e o valor unitário é um número real.

In [ ]:
Produto1 = input('Digite o nome do produto: ')
Quantidade_Vendida1 = int(input('Digite a quantidade vendida: '))
Valor_Unitario1 = float(input('Digite o valor unitário: '))
Valor_Total1 = Quantidade_Vendida1 * Valor_Unitario1
print(Valor_Total1)

Produto2 = input('Digite o nome do produto: ')
Quantidade_Vendida2 = int(input('Digite a quantidade vendida: '))
Valor_Unitario2 = float(input('Digite o valor unitário: '))
Valor_Total2 = Quantidade_Vendida2 * Valor_Unitario2
print(Valor_Total2)

Produto3 = input('Digite o nome do produto: ')
Quantidade_Vendida3 = int(input('Digite a quantidade vendida: '))
Valor_Unitario3 = float(input('Digite o valor unitário: '))
Valor_Total3 = Quantidade_Vendida3 * Valor_Unitario3
print(Valor_Total3)

Faturamento = Valor_Total1 + Valor_Total2 + Valor_Total3
print(Faturamento)


Digite o nome do produto: Boneco Malandrinho
Digite a quantidade vendida: 17
Digite o valor unitário: 18.5
314.5
Digite o nome do produto: Spinner Pequeno
Digite a quantidade vendida: 36
Digite o valor unitário: 12
432.0
Digite o nome do produto: Cubo Mágico
Digite a quantidade vendida: 7
Digite o valor unitário: 5.9
41.300000000000004
787.8


pag 61 - No Exemplo 3.8, vamos escrever um programa que permaneça em laço
enquanto um valor X lido for diferente de zero. Totalize (some todos) e conte os valores digitados, exceto o zero, e apresente esses valores na tela. Use o caso de teste a seguir para verificar se o programa está correto:
:
Entrada Saída
6 8 -30 9 -4 8 16 50 15 7 2 -5 0
# Total dos valores digitados = 82 Quantidade de valores = 12

In [ ]:
cont = 0
soma = 0
while True:
  x = int(input('Digite um valor: '))
  cont += 1
  soma += x
  if x == 0:
    break
print(f'Total dos valores digitados = {soma}')
print(f'Quantidade de valores = {cont}')

Digite um valor: 6
Digite um valor: 8
Digite um valor: -30
Digite um valor: 9
Digite um valor: -4
Digite um valor: 8
Digite um valor: 16
Digite um valor: 50
Digite um valor: 15
Digite um valor: 7
Digite um valor: 2
Digite um valor: -5
Digite um valor: 0
Total dos valores digitados = 82
Quantidade de valores = 13


pag 62 - 1. Altere o programa da PA para, em vez de dez elementos, apresentar na tela uma quantidade Q de elementos, onde Q é um inteiro lido do teclado.


In [ ]:
Primeiro_Termo = int(input('Digite o primeiro termo da PA: '))
Razao = int(input('Digite a razão da PA: '))
Q = int(input('Digite a quantidade de termos: '))
while Q > 0:
  print(Primeiro_Termo)
  Primeiro_Termo += Razao
  Q -= 1


Digite o primeiro termo da PA: 2
Digite a razão da PA: 2
Digite a quantidade de termos: 6
2
4
6
8
10
12


pag 62 - 2. Altere o programa da PA para também calcular e mostrar o somatório de todos os termos. Para isso, crie um novo objeto com o identificador Soma, atribuindo a ele o valor inicial 0, e a cada repetição do laço acrescente a ele um termo da PA.

In [ ]:
Primeiro_Termo = int(input('Digite o primeiro termo da PA:'))
Razao = int(input('Digite a razao da PA:'))
Q = int(input('Digite a quantidade de termos:'))
Soma = Primeiro_Termo

while Q > 0:
  print(Primeiro_Termo)
  Primeiro_Termo += Razao
  Q -= 1
  Soma += Primeiro_Termo

print(f'A soma dos termos é {Soma}')


Digite o primeiro termo da PA:2
Digite a razao da PA:2
Digite a quantidade de termos:6
2
4
6
8
10
12
A soma dos termos é 56


pag 62 - 3. Altere o programa da PA para exibir em cada linha uma frase como a seguir (exemplo: supondo que P = 5, R = 3 e Q = 3). Termo 1 da PA = 5 Termo 2 da PA = 9 Termo 3 da PA = 13


In [ ]:
Primeiro_Termo = int(input('Digite o primeiro termo da PA: '))
Razao = int(input('Digite a razão da PA: '))
Q = int(input('Digite a quantidade de termos: '))

Soma = Primeiro_Termo
Cont = 1

while Q > 0:
  print(f'Termo {Cont} da PA = {Primeiro_Termo}')
  Primeiro_Termo += Razao
  Q -= 1
  Cont += 1
  Soma += Primeiro_Termo

print(f'A soma dos termos é {Soma}')


Digite o primeiro termo da PA: 2
Digite a razão da PA: 2
Digite a quantidade de termos: 6
Termo 1 da PA = 2
Termo 2 da PA = 4
Termo 3 da PA = 6
Termo 4 da PA = 8
Termo 5 da PA = 10
Termo 6 da PA = 12
A soma dos termos é 56


In [ ]:
try:
  Cont = 0
  N = int(input('Digite a quantidade de termos: '))
  A = 0
  B = 1
  print(A)
  print(B)
  while Cont < N - 2:
    C = A + B
    print(C)
    A = B
    B = C
    Cont += 1
except:
  print('Digite um numero inteiro')

Digite a quantidade de termos: 9
0
1
1
2
3
5
8
13
21


PAG 75 - 4. Escreva um programa que leia um número inteiro do teclado e diga se esse número é positivo ou negativo.


In [ ]:
while True:
  try:
    N = int(input('Digite um numero inteiro: '))
  except:
    print('Você não digitou um numero inteiro')
    continue
  else:
    if N > 0:
      print('O numero é positivo')
    elif N < 0:
      print('O numero é negativo')
    else:
      print('O zero é um numero neutro')
  finally:
    Resp = input('Deseja continuar? S/N: ')
    if Resp == 'N' or Resp == 'n':
      break


Digite um numero inteiro: i
Você não digitou um numero inteiro
Deseja continuar? S/N: s
Digite um numero inteiro: 9
O numero é positivo
Deseja continuar? S/N: s
Digite um numero inteiro: -9
O numero é negativo
Deseja continuar? S/N: s
Digite um numero inteiro: -0
O zero é um numero neutro
Deseja continuar? S/N: s
Digite um numero inteiro: -8
O numero é negativo
Deseja continuar? S/N: h
Digite um numero inteiro: u
Você não digitou um numero inteiro
Deseja continuar? S/N: n


pag 75 5. Escreva um programa que leia um número inteiro do teclado e diga se esse número é par ou ímpar. Para saber se um número é par, deve-se verificar se o resto de sua divisão por 2 é igual a zero. Para calcular o resto da divisão de um número por outro deve-se utilizar o operador %. Por exemplo: ao escrever a expressão em negrito a seguir e supondo que A e B tenham conteúdo inteiro.


In [ ]:
try:
  N = int(input('Digite um número inteiro: '))
  if N % 2 == 0:
    print('O número é par')
  else:
    print('O número é ímpar')
except:
  print('Você deve digitar um número inteiro')

Digite um número inteiro: 3
O número é ímpar


pag 76 - 6. Escreva um programa que leia dois números quaisquer e mostre na tela qual é o menor e qual é o maior.

In [ ]:
try:
  N = float(input('Digite um número:'))
  M = float(input('Digite outro número: '))
  if N > M:
    print(f'O maior número é {N}')
    print(f'O menor número é {M}')
  else:
    print(f'O maior número é {M}')
    print(f'O menor número é {N}')
except:
  print('Você deve digitar um número')


Digite um número:6
Digite outro número: 7
O maior número é 7.0
O menor número é 6.0


pag 76 - 7. Escreva um programa que leia três números reais e informe se eles constituem os lados de um triângulo. Em caso afirmativo, informe se o triângulo é equi-látero, isósceles ou escaleno. Para que três números formem um triângulo, a soma dos dois lados menores deve ser maior que o lado maior. Uma boa solução para esse problema envolve o uso dos operadores and e or.

In [ ]:
try:
  L1 = float(input('Digite o primeiro lado do triângulo: '))
  L2 = float(input('Digite o segundo lado do triângulo: '))
  L3 = float(input('Digite o terceiro lado do triângulo: '))
  if L1 < L2 + L3 and L2 < L1 + L3 and L3 < L1 + L2:
    if L1 == L2 == L3:
      print('O triângulo é equilátero')
    elif L1 == L2 or L1 == L3 or L2 == L3:
      print('O triângulo é isósceles')
    else:
      print('O triângulo é escaleno')
  else:
    print('Não é um triângulo')
except:
  print('Você deve digitar um número')


Digite o primeiro lado do triângulo: 4
Digite o segundo lado do triângulo: 5
Digite o terceiro lado do triângulo: 3
O triângulo é escaleno


pag 76 - . Escreva um programa que leia o nome de um lutador e seu peso. Em seguida, informe a categoria a que pertence o lutador, conforme o Quadro 3.9 (note que o quadro foi criado para efeito deste exercício e não condiz com qualquer categoria de luta). A saída do programa deve exibir na tela uma frase com o padrão descrito a seguir: Nome fornecido: Pepe Jordão Peso fornecido: 73.4 Frase a ser exibida: O lutador Pepe Jordão pesa 73,4 kg e se enquadra na categoria Ligeiro
eso Menor que 65 kg
Categoria Pena
Maior ou igual a 65 kg e menor que 72 kg Leve Maior ou igual a 72 kg e menor que 79 kg Ligeiro Maior ou igual a 79 kg e menor que 86 kg Meio-médio Maior ou igual a 86 kg e menor que 93 kg Médio Maior ou igual a 93 kg e menor que 100 kg Meio-pesado Maior ou igual a 100 kg
Pesado


In [ ]:
try:
  Nome = input('Digite o nome do lutador: ')
  Peso = float(input('Digite o peso do lutador: '))
except:
  print('Você deve informar um nome e um peso')
if Peso < 65:
  print(f'O lutador {Nome} pesa {Peso} kg e se enquadra na categoria Pena')
elif Peso >= 65 and Peso < 72:
  print(f'O lutador {Nome} pesa {Peso} kg e se enquadra na categoria Leve')
elif Peso >= 72 and Peso < 79:
  print(f'O lutador {Nome} pesa {Peso} kg e se enquadra na categoria Ligeiro')
elif Peso >= 79 and Peso < 86:
  print(f'O lutador {Nome} pesa {Peso} kg e se enquadra na categoria Meio-médio')
elif Peso >= 86 and Peso < 93:
  print(f'O lutador {Nome} pesa {Peso} kg e se enquadra na categoria Médio')
elif Peso >= 93 and Peso < 100:
  print(f'O lutador {Nome} pesa {Peso} kg e se enquadra na categoria Meio-pesado')
else:
  print(f'O lutador {Nome} pesa {Peso} kg e se enquadra na categoria Pesado')


Digite o nome do lutador: Pepe Jordão
Digite o peso do lutador: 73.4
O lutador Pepe Jordão pesa 73.4 kg e se enquadra na categoria Ligeiro


pag 76 - 9. Escreva um programa que leia o valor hora que um profissional ganha na empresa onde trabalha. Leia também as quantidades de horas normais e horas extras trabalhadas em um mês. Calcule o valor a ser recebido pelo profissional nesse mês, sabendo que nas horas extras o pagamento é dobrado.

In [ ]:
try:
  Sal_hora = float(input('Digite o salário por hora: '))
  Horas_Trabalhadas = float(input('Digite a quantidade de horas trabalhadas: '))
  Horas_Extras = float(input('Digite a quantidade de horas extras trabalhadas: '))
except:
  print('Você deve digitar um número')

Salario_Bruto = Sal_hora * Horas_Trabalhadas
Salario_Extra = Sal_hora * 2 * Horas_Extras
Salario_Total = Salario_Bruto + Salario_Extra
print(f'O salário bruto é {Salario_Bruto}')
print(f'O salário extra é {Salario_Extra}')
print(f'O salário total é {Salario_Total}')

Digite o salário por hora: 10
Digite a quantidade de horas trabalhadas: 10
Digite a quantidade de horas extras trabalhadas: 10
O salário bruto é 100.0
O salário extra é 200.0
O salário total é 300.0


pag 76 10. Escreva um programa que mostre na tela todos os números inteiros de 1 a 10. Para fazer esse programa, será necessário usar o comando de repetição while.

In [ ]:
cont = 1
while cont <= 10:
  print(cont)
  cont += 1

1
2
3
4
5
6
7
8
9
10


pag 76 - 1. Escreva um programa que leia um número inteiro e, em seguida, apresen-te na tela a tabuada de 0 a 10 para esse número fornecido. Siga o formato apresentado a seguir (supondo que foi digitado 4): 4 x 1 = 4 x 2 =
4 8

In [ ]:
N = int(input('Digite um para saber a sua tabuada: '))
cont = 1
while cont <= 10:
  print(f'{N} x {cont} = {N * cont}')
  cont += 1

Digite um para saber a sua tabuada: 4
4 x 1 = 4
4 x 2 = 8
4 x 3 = 12
4 x 4 = 16
4 x 5 = 20
4 x 6 = 24
4 x 7 = 28
4 x 8 = 32
4 x 9 = 36
4 x 10 = 40


pag 77 - 12. Escreva um programa que leia um número inteiro N e, em seguida, leia N números reais, calculando a soma de todos os valores positivos fornecidos e ignorando os negativos. Este exercício pode ser elaborado a partir do Exer-cício resolvido 3.1, no qual, em vez de gerar números aleatórios, os valores sejam lidos do teclado.

In [ ]:
N = int(input('Digite a quantidade de números: '))
cont = 1
soma = 0
while cont <= N:
  num = float(input('Digite um número: '))
  if num > 0:
    soma += num
  cont += 1

print(f' A soma dos números positivos é {soma}')

Digite a quantidade de números: 5
Digite um número: 4
Digite um número: 4
Digite um número: -2
Digite um número: -5
Digite um número: 2
 A soma dos números positivos é 10.0


pag 77 - 13. Escreva um programa que leia valores numéricos inteiros e totalize sepa-radamente os positivos e os negativos até que o usuário digite 0. Ao final, mostre na tela esses dois totais.

In [ ]:
Positivos = 0
Negativos = 0
while True:
  num = int(input('Digite um número inteiro: '))
  if num > 0:
    Positivos += num
  elif num < 0:
    Negativos += num
  else:
    break
print(f'A soma dos números positivos é {Positivos}')
print(f'A soma dos números negativos é {Negativos}')


Digite um número inteiro: 6
Digite um número inteiro: -5
Digite um número inteiro: 6
Digite um número inteiro: -5
Digite um número inteiro: 0
A soma dos números positivos é 12
A soma dos números negativos é -10


pag 77 - 14. Escreva um programa que calcule os N primeiros termos de uma PG com razão R e o primeiro termo P1 fornecidos pelo usuário. Também deve ser calculada e apresentada a soma desses N termos.


In [ ]:
N = int(input('Digite a quantidade de termos: '))
R = float(input('Digite a razão: '))
P1 = float(input('Digite o primeiro termo: '))
cont = 1
soma = P1
print(f'Termo {cont} = {P1}')
while cont < N:
  PG = P1 * R
  cont += 1
  print(f'Termo {cont} = {PG}')
  P1 = PG
  soma += PG
print(f'A soma dos termos é {soma}')

Digite a quantidade de termos: 4
Digite a razão: 2
Digite o primeiro termo: 2
Termo 1 = 2.0
Termo 2 = 4.0
Termo 3 = 8.0
Termo 4 = 16.0
A soma dos termos é 30.0


pag 77 - 15. Escreva um programa que apresente todos os valores inteiros divisíveis por 5 situados no intervalo fechado [Min, Max], em que Min e Max são fornecidos pelo usuário. É obrigatório que o valor Max seja maior que Min e, se isso não ocorrer, o programa deve exibir uma mensagem de aviso ao usuário e inverter os valores.

In [ ]:
Min = int(input('Digite o valor mínimo: '))
Max = int(input('Digite o valor máximo: '))
if Max < Min:
  Min, Max = Max, Min
while Min <= Max:
  if Min % 5 == 0:
    print(Min)
  Min += 1

Digite o valor mínimo: 7
Digite o valor máximo: 30
10
15
20
25
30


pag 77 - 16. Escreva um programa que leia um número inteiro N e, em seguida, leia N números reais, separando o menor e o maior, apresentando-os na tela.


In [ ]:
N = int(input('Digite a quantidade de números: '))
cont = 1
while cont <= N:
  num1 = float(input('Digite um número: '))
  num2 = float(input('Digite outro número: '))
  if num1 > num2:
    print(f'O maior número é {num1} e o menor número é {num2}')
  elif num1 == num2:
    print('Os números são iguais')
  else:
    print(f'O maior número é {num2} e o menor número é {num1}')
  cont += 2
  if cont == N:
    num1 = float(input('Digite um número: '))
    print(f'O maior número é {num1}')
    break





Digite a quantidade de números: 5
Digite um número: 6
Digite outro número: 6
Os números são iguais
Digite um número: 7
Digite outro número: 6
O maior número é 7.0 e o menor número é 6.0
Digite um número: 3
O maior número é 3.0


17. Reescreva um programa do exercício 16 ignorando os números negativos fornecidos pelo usuário.

In [ ]:
N = int(input('Digite a quantidade de números: '))
cont = 1
while cont <= N:
  num1 = float(input('Digite um número: '))
  num2 = float(input('Digite outro número: '))
  if num1 < 0 or num2 < 0:
    print('Você deve digitar números positivos')
    continue
  if num1 > num2:
    print(f'O maior número é {num1} e o menor número é {num2}')
  elif num1 == num2:
    print('Os números são iguais')
  else:
    print(f'O maior número é {num2} e o menor número é {num1}')
  cont += 2
  if cont == N:
    num1 = float(input('Digite um número: '))
    print(f'O maior número é {num1}')
    break


pag 77 - 18. Elabore um programa que efetue a leitura de valores positivos inteiros até que zero ou um valor negativo seja informado. Ao final, devem ser apresentados o maior e menor valores informados pelo usuário, a quantidade de valores, a soma e a média de todos.


In [ ]:
Maior = 0
Menor = 0
Soma = 0
cont = 0

while True:
  try:
    N = int(input('Digite um número inteiro: '))
  except:
    print('Você deve digitar um número inteiro')
  if N <= 0:
    break
  if N > 0:
    if Maior == 0 and Menor == 0:
      Maior = N
      Menor = N
    elif Maior < N:
      Maior = N
    elif Menor > N:
      Menor = N
    Soma += N
    cont += 1
print(f'O maior número é {Maior}')
print(f'O menor número é {Menor}')
print(f'A quantidade de valores digitados foi {cont}')
print(f'A soma dos números é {Soma}')
print(f'A média dos números é {Soma / cont}')

Digite um número inteiro: 10
Digite um número inteiro: 5
Digite um número inteiro: 3
Digite um número inteiro: 0
O maior número é 10
O menor número é 3
A quantidade de valores digitados foi 3
A soma dos números é 18
A média dos números é 6.0


 Escreva um programa que contenha um laço que será executado enquanto o número digitado for diferente de zero. Para cada número digitado pelo usuário, mostrar na tela apenas os que forem divisíveis por 2 e por 3.


In [ ]:
try:
  while True:
    N = int(input('Digite um número inteiro: '))
    if N == 0:
      break
    if N % 2 == 0 and N % 3 == 0:
      print(N)
except:
  print('Você deve digitar um número inteiro')

Digite um número inteiro: 10
Digite um número inteiro: 12
12
Digite um número inteiro: 0


20. Elabore um programa que apresente o somatório dos valores pares exis-tentes na faixa entre 1 e N, em que N é um número digitado pelo usuário e que deve ser no mínimo 100 (obrigatório garantir esse requisito).


In [ ]:
cont = 0
soma = 0
try:
  while True:
    N = int(input('Digite um número maior que 100:'))
    if N <= 100:
      print('Número digitado menor que 100')
      continue
    else:
      break
  while cont <= N:
    if cont % 2 == 0:
      soma += cont
    cont += 1
  print(f'O somatório dos valores pares foi de {soma}')
except:
  print('Você deve digitar um número inteiro')



Digite um número maior que 100:99
Número digitado menor que 100
Digite um número maior que 100:103
O somatório dos valores pares foi de 2652


21. Reescreva o programa do Exercício resolvido 3.2 – Sequência de Fibonacci fazendo a seguinte alteração: leia N que será a quantidade de termos a ser exibida e leia um número inteiro adicional chamado Prim. Essa versão do programa deverá apresentar N termos da sequência de Fibonacci imedia-tamente maiores que Prim.


In [ ]:
N = int(input('Digite a quantidade de termos: '))
Prim = int(input('Digite um número inteiro: '))
cont = 1
A = 0
B = 1
if Prim < A:
  print(A)
if Prim < B:
  print(B)
while cont <= N - 2:
  C = A + B
  if C > Prim:
    print(C)
  A = B
  B = C
  cont += 1

Digite a quantidade de termos: 5
Digite um número inteiro: -1
0
1
1
2
3


pg 94 - 5. Criação de uma lista contendo números inteiros Escreva um programa que leia um número inteiro N e gere uma lista com
N elementos aleatórios (utilize a função randint explicada no Exercício resolvido 3.7) entre 10 e 50. Exiba a lista gerada e exiba também a mesma lista com seus elementos ordenados em ordem crescente.

In [ ]:
from random import randint

try:
  N = int(input('Digite a quantidade de números: '))
except:
  print('Você deve digitar um número inteiro')

Lista = []
for i in range(N):
  Lista.append(randint(10, 50))
Lista.sort()
print(Lista)

Digite a quantidade de números: 7
[11, 12, 14, 15, 35, 36, 41]


pag 95 - Como foi visto, a execução desse programa para N = 8 gerou uma lista con-tendo duas ocorrências do valor 50. Nada foi dito no enunciado sobre isso. A tarefa é alterar esse programa de modo que não haja valores duplicados na lista gerada. Dica: lembre-se dos operadores in e not in.

In [ ]:
from random import randint
try:
  N = int(input('Digite a quantidade de números: '))
except:
  print('Você deve digitar um número inteiro')
Lista = []
Count = 0
while Count < N:
  Num = randint(10, 50)
  if Num not in Lista:
    Lista.append(Num)
    Count += 1
Lista.sort()
print(Lista)


Digite a quantidade de números: 6
[12, 28, 31, 37, 40, 41]


pag - 116 -1. Escreva um programa que leia do teclado uma lista com tamanho de 10 elementos e exiba-a na tela na ordem inversa à ordem de leitura.


In [ ]:
L = []
while len(L) < 10:
  try:
    L.append(int(input('Digite um número inteiro: ')))
  except:
    print('Você deve digitar um número inteiro')
L.reverse()
print(L)


Digite um número inteiro: 1
Digite um número inteiro: 2
Digite um número inteiro: 3
Digite um número inteiro: 4
Digite um número inteiro: t
Você deve digitar um número inteiro
Digite um número inteiro: 5
Digite um número inteiro: 6
Digite um número inteiro: u
Você deve digitar um número inteiro
Digite um número inteiro: 7
Digite um número inteiro: 8
Digite um número inteiro: 9
Digite um número inteiro: 10
[10, 9, 8, 7, 6, 5, 4, 3, 2, 1]


pag 116 - 2. Escreva um programa que leia do teclado duas listas com tamanho 10, com números inteiros. Em seguida, o programa deve juntar as duas listas em uma única com o tamanho 20.

In [ ]:
L1 = []
L2 = []

Cont = 0
while Cont < 20:
  try:
    if Cont < 10:
      L1.append(int(input('Digite um número inteiro: ')))
      Cont += 1
    else:
      L2.append(int(input('Digite um número inteiro: ')))
      Cont += 1
  except:
    print('Você deve digitar um número inteiro')

L3 = L1 + L2
print(L3)


Digite um número inteiro: 1
Digite um número inteiro: 2
Digite um número inteiro: 3
Digite um número inteiro: 4
Digite um número inteiro: 5
Digite um número inteiro: 6
Digite um número inteiro: 7
Digite um número inteiro: 8
Digite um número inteiro: 9
Digite um número inteiro: 10
Digite um número inteiro: 11
Digite um número inteiro: 12
Digite um número inteiro: 13
Digite um número inteiro: 14
Digite um número inteiro: 15
Digite um número inteiro: 16
Digite um número inteiro: 17
Digite um número inteiro: 18
Digite um número inteiro: r
Você deve digitar um número inteiro
Digite um número inteiro: t
Você deve digitar um número inteiro
Digite um número inteiro: 19
Digite um número inteiro: 20
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]


pag 116 - 3. Escreva um programa que preencha com números inteiros duas listas de-nominadas A e B com diferentes tamanhos nA e nB, respectivamente. Em seguida, o programa deve juntar as duas em uma única lista com o tamanho nA + nB. Exibir na tela a lista resultante.

In [ ]:
A = []
B = []
while True:
  try:
    NA = int(input('Digite o tamanho da lista A: '))
    if NA <= 0:
      print('Você deve digitar um número maior que 0')
      continue
    else:
      break
  except:
    print('Você deve digitar um número inteiro')
while True:
  try:
    NB = int(input('Digite o tamanho da lista B: '))
    if NB <= 0:
      print('Você deve digitar um número maior que 0')
      continue
    else:
      break
  except:
    print('Você deve digitar um número inteiro')

Cont = 0
while Cont < (NA + NB):
  try:
    if Cont <= NA:
      A.append(int(input('Digite um numero inteiro: ')))
      Cont += 1
    else:
      B.append(int(input('Digite um número inteiro: ')))
      Cont += 1
  except:
    print('Você deve digitar um número inteiro')

C = A + B
print(C)



Digite o tamanho da lista A: e
Você deve digitar um número inteiro
Digite o tamanho da lista A: 3
Digite o tamanho da lista B: 5
Digite um numero inteiro: 1
Digite um numero inteiro: 2
Digite um numero inteiro: 3
Digite um numero inteiro: e
Você deve digitar um número inteiro
Digite um numero inteiro: 4
Digite um número inteiro: 5
Digite um número inteiro: 6
Digite um número inteiro: 7
Digite um número inteiro: 8
[1, 2, 3, 4, 5, 6, 7, 8]


pag 116 - 4. Escreva um programa que leia uma lista com N números inteiros, em que N é um número inteiro previamente digitado pelo usuário. O programa não deve aceitar um número digitado que já esteja inserido na lista, sendo que, quando essa situação ocorrer, uma mensagem deve ser dada ao usuário. Por fim, exibir na tela a lista resultante.

In [ ]:
L = []

while True:
  try:
    N = int(input('Digite a quantidade de números: '))
    if N <= 0:
      print('Você deve digitar um número maior que 0')
      continue
    else:
      break
  except:
    print('Você deve digitar um número inteiro')
while len(L) < N:
  try:
    Num = int(input('Digite um número inteiro: '))
    if Num in L:
      print('Você ja digitou esse número')
      continue
    else:
      L.append(Num)
  except:
    print('Você deve digitar um número inteiro')

print(L)

Digite a quantidade de números: r
Você deve digitar um número inteiro
Digite a quantidade de números: -2
Você deve digitar um número maior que 0
Digite a quantidade de números: 5
Digite um número inteiro: 4
Digite um número inteiro: t
Você deve digitar um número inteiro
Digite um número inteiro: -5
Digite um número inteiro: 3
Digite um número inteiro: 6
Digite um número inteiro: 2
[4, -5, 3, 6, 2]


pag 116 - 5. Escreva um programa que leia do teclado dois números inteiros nA e nB e leia também duas listas denominadas A e B com os tamanhos nA e nB, respectivamente. Na leitura de cada uma das listas é obrigatório que não sejam aceitos valores repetidos. Em seguida, o programa deve juntar as duas em uma única lista R (resultante), tomando o cuidado de que R não tenha valores duplicados.

In [ ]:
while True:
  try:
    nA = int(input('Digite o tamanho da lista A: '))
    if nA <= 0:
      print('Você deve digitar um número maior que 0')
      continue
    else:
      break
  except:
    print('Você deve digitar um número inteiro')
while True:
  try:
    nB = int(input('Digite o tamanho da lista B: '))
    if nB <= 0:
      print('Você deve digitar um número maior que 0')
      continue
    else:
      break
  except:
    print('Você deve digitar um número inteiro')

A = []
B = []
R = []

cont = 0

while cont < nA:
  try:
    numA = (int(input('Digite um número inteiro: ')))
    if numA not in A:
      A.append(numA)
      cont += 1
    else:
      print('Você ja digitou esse número')
  except:
    print('Você deve digitar um número inteiro')

cont = 0
while cont < nB:
  try:
    numB = (int(input('Digite um número inteiro: ')))
    if numB not in B:
      B.append(numB)
      cont += 1
    else:
      print('Você ja digitou esse número')
  except:
    print('Você deve digitar um número inteiro')

R = A.copy()
for i in B:
  if i not in R:
    R.append(i)
print(R)





Digite o tamanho da lista A: 3
Digite o tamanho da lista B: 4
Digite um número inteiro: 1
Digite um número inteiro: 1
Você ja digitou esse número
Digite um número inteiro: e
Você deve digitar um número inteiro
Digite um número inteiro: 2
Digite um número inteiro: 3
Digite um número inteiro: 2
Digite um número inteiro: 3
Digite um número inteiro: e
Você deve digitar um número inteiro
Digite um número inteiro: 3
Você ja digitou esse número
Digite um número inteiro: 4
Digite um número inteiro: 5
[1, 2, 3, 4, 5]


pag 116 -  6. Escreva um programa que leia três dados de entrada: o primeiro termo, a razão e a quantidade de termos de uma P.A., todos números inteiros. O programa deve calcular todos os termos, colocando-os em uma lista, e exi-bi-la no final. Esse exercício já foi resolvido e explicado no Capítulo 3 (veja Exercício resolvido 3.2). A diferença, aqui, é que se pede para utilizar uma lista para armazenar os diversos termos antes de exibi-los.

In [ ]:
try:
  L = [int(input('Digite o primeiro termo: '))]
  R = int(input('Digite a razão: '))
  N = int(input('Digite a quantidade de termos: '))
except:
  print('Você deve digitar um número inteiro')

for i in range(N - 1):
  L.append(L[-1] + R)
print(L)


Digite o primeiro termo: 2
Digite a razão: 2
Digite a quantidade de termos: 4
[2, 4, 6, 8]


pag 117 - 7. Escreva um programa que leia um número N obrigatoriamente entre 0 e 50 e, em seguida, leia N números reais em uma lista A. O programa deve separar os valores lidos em A em outras duas listas NEG e POS: a primeira contendo somente os valores negativos e a segunda contendo os valores positivos e zero. Apresentar na tela as listas NEG e POS e a quantidade de valores contidos em cada uma.


In [ ]:
while True:
  try:
    N = int(input('Digite a quantidade de números: '))
    if N < 0 or N > 50:
      print('Você deve digitar um número entre 0 e 50')
    else:
      break
  except:
    print('Você deve digitar um número inteiro')

A = []
NEG = []
POS = []
while len(A) < N:
  try:
    A.append(float(input('Digite um número: ')))
    if A[-1] < 0:
      NEG.append(A[-1])
    else:
      POS.append(A[-1])
  except:
    print('Você deve digitar um número')

print(f'A lista A é {A}')
print(f'A lista NEG é {NEG}')
print(f'A lista POS é {POS}')





Digite a quantidade de números: r
Você deve digitar um número inteiro
Digite a quantidade de números: -1
Você deve digitar um número entre 0 e 50
Digite a quantidade de números: 60
Você deve digitar um número entre 0 e 50
Digite a quantidade de números: 6
Digite um número: -1
Digite um número: 0
Digite um número: r
Você deve digitar um número
Digite um número: 4
Digite um número: 3
Digite um número: 7
Digite um número: 8
A lista A é [-1.0, 0.0, 4.0, 3.0, 7.0, 8.0]
A lista NEG é [-1.0]
A lista POS é [0.0, 4.0, 3.0, 7.0, 8.0]


pag 117 - 8. Escreva um programa que leia um número N (entre 0 e 50) e, em seguida, defina uma lista V preenchendo-a com N números inteiros aleatórios (utilizar a função randint). Exiba-a na tela. Inicie um laço no qual será feita a leitura de um número X e que termina quando X for zero. Pesquise se X está ou não na lista V e, caso esteja, elimine todas as suas ocorrências.


In [ ]:
from random import randint

while True:
  try:
    N = int(input('Digite a quantidade de números: '))
    if N < 0 or N > 50:
      print('Você deve digitar um número entre 0 e 50')
    else:
      break
  except:
    print('Você deve digitar um número inteiro')

V = []
for i in range(N):
  V.append(randint(0, 100))

print(V)

cont = 0
X = int(input('Digite um número: '))
for i in V:
  if i == X:
    V.remove(X)

print(V)






Digite a quantidade de números: 8
[66, 14, 37, 46, 42, 64, 41, 29]
Digite um número: 66
[14, 37, 46, 42, 64, 41, 29]


pag 117 - 9. O programa deverá ler dois inteiros chamados Min e Max. Min pode ser qualquer valor e Max, obrigatoriamente, deve ser maior que Min. Em segui-da, preencher uma lista com todos os valores divisíveis por 7 contidos no intervalor fechado [Min, Max]. Exibir a lista resultante na tela.

In [ ]:
while True:
  try:
    min = int(input('Digite um número inteiro: '))
    max = int(input('Digite um número inteiro maior que o anterior: '))
    if max <= min:
      print('Você deve digitar um número maior que o anterior')
    else:
      break
  except:
   print('Você deve digitar um número inteiro')

L = []
for i in range(min, max + 1):
  if i % 7 == 0:
    L.append(i)
print(L)


Digite um número inteiro: 3
Digite um número inteiro maior que o anterior: 20
[7, 14]


pag 117 - 10. Escreva um programa que leia do teclado uma lista com N elementos. Em seguida, o programa deve eliminar os elementos que estiverem repetidos, mantendo apenas a primeira ocorrência de cada. Apresentar a lista resul-tante na tela. Os valores eliminados devem ser armazenados em outra lista que também deve ser exibida.


In [ ]:
while True:
  try:
    N = int(input('Digite a quantidade de números: '))
    if N <= 0:
      print('Você deve digitar um número maior que 0:')
    else:
      break
  except:
    print('Você deve digitar um número inteiro')

cont = 0
L = []
while cont < N:
  try:
    L.append(int(input('Digite um número inteiro: ')))
    cont += 1
  except:
    print('Você deve digitar um número inteiro')

L2 = []
for i in L:
  if i not in L2:
    L2.append(i)
print(L2)
print(L)

Digite a quantidade de números: 5
Digite um número inteiro: 5
Digite um número inteiro: 6
Digite um número inteiro: 6
Digite um número inteiro: 7
Digite um número inteiro: 8
[5, 6, 7, 8]
[5, 6, 6, 7, 8]


pag 117 - 11. Faça um programa que leia um número inteiro N bem grande (acima de 5.000). Preencha uma lista de tamanho N com números inteiros aleatórios positivos. Em seguida, inicie um laço de pesquisa, no qual o valor a ser pesquisado deve ser lido do teclado, e o programa deve dizer se tal valor está ou não contido na lista, bem como dizer sua posição. No caso de várias ocorrências, exibir todas. O laço de pesquisa termina quando for digitado o zero. Use o algoritmo de busca sequencial.

In [ ]:
from random import randint

while True:
  try:
    N = int(input('Você deve digitar um número maior que 5000: '))
    if N <= 5000:
      print('')
    else:
      break
  except:
    print('Você deve digitar um número inteiro')

L = []
for i in range(N):
  L.append(randint(0, 5000))

while True:
  try:
    X = int(input('Digite um número inteiro: '))
    if X == 0:
      break
    else:
      for i in L:
        if i == X:
          print(f'O número {X} está na lista e está na posição {L.index(X)}')
          continue
      if X not in L:
        print(f'O número {X} não está na lista')
  except:
    print('Você deve digitar um número inteiro')


Você deve digitar um número maior que 5000: 8

Você deve digitar um número maior que 5000: 5001
Digite um número inteiro: 5
O número 5 está na lista e está na posição 4645
Digite um número inteiro: 6
O número 6 não está na lista
Digite um número inteiro: 667
O número 667 não está na lista
Digite um número inteiro: t
Você deve digitar um número inteiro
Digite um número inteiro: 0


pag 117 - 12. Escreva um programa que leia do teclado duas matrizes de dimensões 2×2 e mostre na tela a soma dessas duas matrizes.

In [ ]:
M1 = []
M2 = []
M3 = []

for i in range(4):
  M1.append(int(input('Digite um número inteiro: ')))
for i in range(4):
  M2.append(int(input('Digite um número inteiro: ')))

M3.append(M1[0] + M2[0])
M3.append(M1[1] + M2[1])
M3.append(M1[2] + M2[2])
M3.append(M1[3] + M2[3])

print(M3)


Digite um número inteiro: 4
Digite um número inteiro: 5
Digite um número inteiro: 6
Digite um número inteiro: 7
Digite um número inteiro: 8
Digite um número inteiro: 9
Digite um número inteiro: 1
Digite um número inteiro: 2
[12, 14, 7, 9]


pag 117 13. Escreva um programa que leia do teclado duas matrizes de dimensões 2×2 e mostre na tela a multiplicação dessas duas matrizes.


In [ ]:
M1 = []
M2 = []
M3 = []

for i in range(4):
  M1.append(int(input('Digite um número inteiro: ')))
for i in range(4):
  M2.append(int(input('Digite um número inteiro: ')))

M3.append(M1[0] * M2[0])
M3.append(M1[1] * M2[1])
M3.append(M1[2] * M2[2])
M3.append(M1[3] * M2[3])

print(M3)

Digite um número inteiro: 4
Digite um número inteiro: 4
Digite um número inteiro: 4
Digite um número inteiro: 4
Digite um número inteiro: 4
Digite um número inteiro: 4
Digite um número inteiro: 4
Digite um número inteiro: 4
[16, 16, 16, 16]


pag 117 14. A matriz a seguir mostra o custo unitário de cada produto e a quantidade de cada um dos produtos no estoque de três lojas de uma rede. Escreva um programa que exiba na tela as respostas para as perguntas. Na solução desse problema, elabore uma maneira de armazenar seus dados utilizando lista e sublistas. Os dados da matriz devem ser lidos do teclado.

a) Qual é o valor total de estoque em cada uma das lojas? b) Qual é o valor total de estoque para cada produto disponível na rede? c) Qual é o valor total de estoque da rede

In [ ]:
L1 = []
L2 = []
L3 = []

while True:
  try:
    N = int(input('Qual loja deseja acessar \n loja 1 \n loja 2 \n loja 3 \n 0 para sair \n '))
    if N == 0:
      break
    elif N == 1:
      for i in range(4):
        produto = (str(input('Digite o nome do produto: ')))
        custo = (float(input('Digite o custo do produto: ')))
        qtda = (int(input('Digite a quantidade do produto: ')))
        L1.append([produto, custo, qtda])
    elif N == 2:
      for i in range(4):
        produto = (str(input('Digite o nome do produto: ')))
        custo = (float(input('Digite o custo do produto: ')))
        qtda = (int(input('Digite a quantidade do produto: ')))
        L2.append([produto, custo, qtda])
    elif N == 3:
      for i in range(4):
        produto = (str(input('Digite o nome do produto: ')))
        custo = (float(input('Digite o custo do produto: ')))
        qtda = (int(input('Digite a quantidade do produto: ')))
        L3.append([produto, custo, qtda])
    elif N == 4:
      for i in range(4):
        produto = (str(input('Digite o nome do produto: ')))
        custo = (float(input('Digite o custo do produto: ')))
        qtda = (int(input('Digite a quantidade do produto: ')))
        L4.append([produto, custo, qtda])
  except:
    print('Digite corretamente os dados')
    continue

Total1 = 0
for i in range(1):
  for j in range(4):
    Total1 += L1[j][1] * L1[j][2]
  print(f'O valor total de estoque na loja 1 é R$ {Total1}')

Total2 = 0
for i in range(1):
  for j in range(4):
    Total2 += L1[j][1] * L1[j][2]
  print(f'O valor total de estoque na loja 2 é R$ {Total2}')

Total3 = 0
for i in range(1):
  for j in range(4):
    Total3 += L1[j][1] * L1[j][2]
  print(f'O valor total de estoque na loja 3 é R$ {Total3}')


T_prodEstoqueA = (L1[0][2] + L2[0][2] + L3[0][2]) * L1[0][1]
print(f'O valor total de estoque para cada Produto A disponível na rede é {T_prodEstoqueA}')

T_prodEstoqueB = (L1[1][2] + L2[1][2] + L3[1][2]) * L1[1][1]
print(f'O valor total de estoque para cada Produto B disponível na rede é {T_prodEstoqueB}')

T_prodEstoqueC = (L1[2][2] + L2[2][2] + L3[2][2]) * L1[2][1]
print(f'O valor total de estoque para cada Produto C disponível na rede é {T_prodEstoqueC}')

T_prodEstoqueD = (L1[3][2] + L2[3][2] + L3[3][2]) * L1[3][1]
print(f'O valor total de estoque para cada Produto D disponível na rede é {T_prodEstoqueD}')

TotalEstoque = T_prodEstoqueA + T_prodEstoqueB + T_prodEstoqueC + T_prodEstoqueD
print(f'O valor total de estoque da rede é {TotalEstoque}')



Qual loja deseja acessar 
 loja 1 
 loja 2 
 loja 3 
 0 para sair 
 1
Digite o nome do produto: a
Digite o custo do produto: 1
Digite a quantidade do produto: 2
Digite o nome do produto: b
Digite o custo do produto: 3
Digite a quantidade do produto: 4
Digite o nome do produto: c
Digite o custo do produto: 5
Digite a quantidade do produto: 6
Digite o nome do produto: d
Digite o custo do produto: 7
Digite a quantidade do produto: 8
Qual loja deseja acessar 
 loja 1 
 loja 2 
 loja 3 
 0 para sair 
 2
Digite o nome do produto: a
Digite o custo do produto: 1
Digite a quantidade do produto: 2
Digite o nome do produto: b
Digite o custo do produto: 3
Digite a quantidade do produto: 4
Digite o nome do produto: c
Digite o custo do produto: 5
Digite a quantidade do produto: 6
Digite o nome do produto: d
Digite o custo do produto: 7
Digite a quantidade do produto: 8
Qual loja deseja acessar 
 loja 1 
 loja 2 
 loja 3 
 0 para sair 
 3
Digite o nome do produto: a
Digite o custo do produto: 1
Digit

In [ ]:
def Fatorial(N):
  if N <= 1:
   return 'fim'

  else:
    print(N)
    return N * Fatorial(N-1)


X = int(input('Digite N: '))
F = Fatorial(X)


Digite N: 6
6
5
4
3
2


pag 139 - 1. Escreva uma função que recebe um número inteiro como parâmetro de entrada e retorna o texto “PAR” ou “ÍMPAR”


In [ ]:
def par_impar(N):
  if N % 2 == 0:
    return 'Par'
  else:
    return 'Impar'

par_impar(4)

'Par'

pag 139 2. Utilize a função EPrimo desenvolvida no Exercício resolvido 5.1 para carregar uma lista contendo os N primeiros números primos, em que N é um número inteiro fornecido pelo usuário.

In [ ]:
def EPrimo(P):
  if P <= 1:
    return 'Erro'
  elif P == 2:
    return 1
  elif P % 2 == 0:
    return 0
  else:
    raiz = P ** 0.5
    R = 1
    i = 3
    while i <= raiz and R != 0:
      R = P % i
      i+=2
    return R

L = []
Termo = 2
N = int(input('Digite a quantidade de números primos da lista: '))
while True:
  if len(L) == N:
    break
  elif EPrimo(Termo) == 1:
    L.append(Termo)
    Termo += 1
  else:
    Termo += 1

print(L)


Digite a quantidade de números primos da lista: 20
[2, 3, 5, 7, 13, 19, 31, 41, 71, 109, 241, 271, 307, 419, 463, 599, 701, 757, 811, 929]


pag 139 3. Escreva uma função que receba dois números inteiros A e B como parâme-tros de entrada e retorne 1 se A for divisível por B e 0 caso contrário.


In [ ]:
def divisivel(A, B):
  if A % B == 0:
    return 1
  else:
    return 0

divisivel(10, 2)


1

pag 139 4. Escreva uma função que receba um número inteiro N e retorne uma lista com os bits 0 e 1, que representam N convertido para binário. Não use nenhuma função Python de conversão para binários. Em vez disso, elabore uma lógica baseada no processo de divisões sucessivas.


In [ ]:
def binario(N):
  L = []
  while N > 0:
    L.append(N % 2)
    N = N // 2
  L.reverse()
  return L

binario(14)

[1, 1, 1, 0]

pag 139 . Escreva uma função que receba como parâmetro de entrada uma lista L de números inteiros e um valor. A função deve retornar quantas vezes o valor está contido na lista. Caso ele não esteja em L, retorne 0.

In [ ]:
def contagem(L, V):
  "Recebe uma lista e um valor e retorna a quantidade de vezes que o valor está contido na lista"
  cont = 0
  for i in L:
    if i == V:
      cont += 1
  return cont

contagem([1, 2, 3, 4, 5], 3)

1

pag 139 Escreva um programa que receba como parâmetro de entrada um número inteiro de 5 dígitos no intervalo fechado [10000, 30000] que represente có-digos de produtos vendidos em uma loja. A função deve calcular e retornar o dígito verificador utilizando regra de cálculo explicada a seguir. Considere o código 21853, em que cada dígito é multiplicado por um peso começando em 2, os valores obtidos são somados, e do total obtido calcula-se o resto de sua divisão por 7.

In [ ]:
def digito_verificador(N):
  Soma = 0
  Peso = 2
  for i in N:
    Soma = Soma + int(i) * Peso
    Peso += 1
  return Soma % 7

try:
  while True:
    N = input('Digite um número no intervalo entre 10000, 30000: ')
    if int(N) < 10000 or int(N) > 30000:
      print('Você deve digitar um número entre 10000 e 30000')
    else:
      break
except:
  print('Você deve digitar um número inteiro')

print(digito_verificador(N))







Digite um número no intervalo entre 10000, 30000: 21853
5


pag 139 7. Escreva uma função que receba como parâmetro de entrada dois números reais Min e Max. Essa função deve ler do teclado um número real e retorná-lo caso esteja dentro do intervalo fechado [Min, Max]. Caso contrário, a função deve exibir uma mensagem de erro e ler um novo valor.

In [ ]:
def intervalo(Min, Max):
  while True:
    N = float(input('Digite um número real para verificar se está no intervalo: '))
    if N >= Min and N <= Max:
      return N
      break
    else:
      print('Você deve digitar um número real dentro do intervalo')

try:
  while True:
    Min = float(input('Digite um número real: '))
    Max = float(input('Digite um número real maior que o anterior: '))
    if Max <= Min:
      print('Você deve digitar um número real maior que o anterior')
    else:
      break
except:
  print('Você deve digitar um número real')

intervalo(Min, Max)



Digite um número real: 4
Digite um número real maior que o anterior: 9
Digite um número real para verificar se está no intervalo: 3
Você deve digitar um número real dentro do intervalo
Digite um número real para verificar se está no intervalo: 6


6.0

pag 139 - 8. Escreva uma função que receba uma lista como parâmetro de entrada e retorne uma tupla contendo quatro valores na seguinte ordem: a soma, a média, o menor e o maior valor dentre todos os elementos nela contidos. Considere que nessa lista ocorram apenas números reais. Escreva um programa para testar essa função, exibindo na tela os resultados. Neste exercício, evite utilizar as funções prontas existentes no Python, como sum, min e max.

In [ ]:
def calculos(L):
  soma = 0
  media = 0
  menor = L[0]
  maior = L[0]
  for i in L:
    soma += i
    media = soma / len(L)
    if i < menor:
      menor = i
    elif i > maior:
      maior = i

  resultado = (soma, media, menor, maior)
  return resultado

calculos([1, 2, 3, 4, 5])


(15, 3.0, 1, 5)

pag 140 9. Escreva uma função que receba uma lista L e elimine os eventuais elemen-tos repetidos contidos na mesma, deixando na lista resultante apenas uma ocorrência de cada elemento. Escreva um programa para testar essa fun-ção, o qual deve ler do teclado os elementos que farão parte da lista. (veja o Exercício proposto 4.10)

In [ ]:
def eliminar_repetidos(L):
  L2 = []
  for i in L:
    if i not in L2:
      L2.append(i)
  return L2

L = []
try:
  while True:
    E = input('Digite um elemento para a lista ou espaço para sair: ')
    if E == ' ':
      break
    else:
      L.append(E)
except:
  print('Você deve digitar um elemento para a lista')

eliminar_repetidos(L)


Digite um elemento para a lista ou espaço para sair: 5
Digite um elemento para a lista ou espaço para sair: casa
Digite um elemento para a lista ou espaço para sair: 0
Digite um elemento para a lista ou espaço para sair: bola
Digite um elemento para a lista ou espaço para sair: a
Digite um elemento para a lista ou espaço para sair: bola
Digite um elemento para a lista ou espaço para sair:  


['5', 'casa', '0', 'bola', 'a']

pag 140 10. Escreva uma função que receba duas listas L1 e L2 como parâmetro de en-trada e retorne uma lista contendo todos os elementos de L1 que não estão em L2. Escreva um programa para testar essa função.


In [ ]:
def diferenca(L1, L2):
  L3 = []
  for i in L1:
    if i not in L2:
      L3.append(i)
  return L3

L1 = []
L2 = []

while True:
  E = input('Digite um elemento para a lista 1 ou espaço para sair: ')
  if E == ' ':
    break
  else:
    L1.append(E)

while True:
  E = input('Digite um elemento para a lista 2 ou espaço para sair: ')
  if E == ' ':
    break
  else:
    L2.append(E)

diferenca(L1, L2)

Digite um elemento para a lista 1 ou espaço para sair: casa
Digite um elemento para a lista 1 ou espaço para sair: 2
Digite um elemento para a lista 1 ou espaço para sair: batata
Digite um elemento para a lista 1 ou espaço para sair: 56
Digite um elemento para a lista 1 ou espaço para sair: casa 1
Digite um elemento para a lista 1 ou espaço para sair:  
Digite um elemento para a lista 2 ou espaço para sair: casa
Digite um elemento para a lista 2 ou espaço para sair: r
Digite um elemento para a lista 2 ou espaço para sair: bola
Digite um elemento para a lista 2 ou espaço para sair:  


['2', 'batata', '56', 'casa 1']

pag 140 11. Escreva uma função que receba como parâmetro de entrada uma lista L e retorne uma lista organizada em ordem crescente. Para fazer a ordenação, use o Algoritmo de Ordenação Bolha (Bubble Sort). Crie uma segunda versão dessa função que retorne uma lista organizada em ordem decrescente. Escreva um programa para testar essas duas funções. Esse programa deve ler um número inteiro N e gerar uma lista com N números inteiros aleatórios utilizando a função randint(). Use as duas funções de ordenação e exiba na tela as listas ordenadas crescente e decrescente.


In [ ]:
from random import randint

def bolha(L):
  for i in range(len(L)):
    for j in range(len(L)-1):
      if L[j] > L[j+1]:
        L[j], L[j+1] = L[j+1], L[j]
  return L

def bolha_decrescente(L):
  for i in range(len(L)):
    for j in range(len(L)-1):
      if L[j] < L[j+1]:
        L[j], L[j+1] = L[j+1], L[j]
  return L

N = int(input('Digite um número inteiro: '))
L = []
for i in range(N):
  L.append(randint(0, 100))

print(bolha(L))
print(bolha_decrescente(L))


Digite um número inteiro: 6
[6, 12, 14, 14, 33, 74]
[74, 33, 14, 14, 12, 6]


pag 140 12. Escreva um programa que leia um número inteiro Q e exiba na tela os Q primeiros termos da sequência de Fibonacci, utilizando uma função recur-siva para determinar o elemento da sequência a ser exibido. A sequência de Fibonacci, já vista nos Exercícios resolvidos 3.7 e 4.7, caracteriza-se por um termo ser a soma dos dois anteriores, sendo que os dois primeiros termos são 0 e 1. Assim, os dez primeiros termos são: 0, 1, 1, 2, 3, 5, 8, 13, 21, 34.

In [ ]:
def fibonacci(Q):
  L = [0, 1]
  for i in range(Q-2):
    L.append(L[i] + L[i+1])
  return L
  if N <= 1:
    return N


Q = int(input('Digite um número inteiro: '))
fibonacci(Q)

Digite um número inteiro: 10


[0, 1, 1, 2, 3, 5, 8, 13, 21, 34]

pag 140 13. Dada uma lista contendo números inteiros, escreva uma função recursiva para calcular a multiplicação de todos os elementos. Exiba o resultado na tela.

In [ ]:
def multiplicacao(L):
  if len(L) == 1:
    return L[0]
  else:
    return L[0] * multiplicacao(L[1:])

L = [1, 2, 3, 4, 5]
multiplicacao(L)

120

pag 140 14. Faça uma pesquisa sobre o Algoritmo de Ordenação Quicksort. Implemente uma função recursiva que use esse algoritmo para organizar a lista L de forma crescente. Escreva um programa para testar a função.

In [ ]:
def quicksort(L):
  if len(L) <= 1:
    return L
  else:
    pivo = L[0]
    menores = [x for x in L[1:] if x <= pivo]
    maiores = [x for x in L[1:] if x > pivo]
    return quicksort(menores) + [pivo] + quicksort(maiores)

L = [1, 4, 3, 5, 2]
quicksort(L)

[1, 2, 3, 4, 5]

pag 150 5. Escreva um programa que contenha duas funções de ordenação diferentes: uma que implemente o algoritmo Bubble Sort (criada no Exercício 11) e outra que implemente o algoritmo Quicksort (criada no Exercício 14). Escreva um programa que gere uma lista com um tamanho bem grande e utilize-a para testar o desempenho das duas funções. Sugestões: Faça esse programa gerar a lista grande contendo Q elementos, na qual Q é digitado pelo usuário. Esse valor de Q deve ser bem grande, por exemplo, de 10 a 50 mil elementos, ou mais, dependendo da capacidade de seu computador.

In [ ]:
from random import randint

def bolha(L):
  for i in range(len(L)):
    for j in range(len(L)-1):
      if L[j] > L[j+1]:
        L[j], L[j+1] = L[j+1], L[j]
  return L

def quicksort(L):
  if len(L) <= 1:
    return L
  else:
    pivo = L[0]
    menores = [x for x in L[1:] if x <= pivo]
    maiores = [x for x in L[1:] if x > pivo]
    return quicksort(menores) + [pivo] + quicksort(maiores)


N = int(input('Digite um número inteiro: '))
L = []
for i in range(N):
  L.append(randint(0, 10000))

print(bolha(L))
print(quicksort(L))


Digite um número inteiro: 1000
[15, 32, 63, 89, 102, 142, 143, 154, 177, 201, 213, 216, 227, 232, 233, 247, 275, 284, 284, 285, 308, 318, 338, 339, 341, 342, 374, 379, 381, 391, 395, 398, 402, 407, 422, 422, 423, 428, 437, 440, 442, 449, 459, 461, 486, 492, 495, 502, 504, 505, 505, 517, 521, 541, 546, 555, 555, 556, 596, 597, 612, 620, 645, 649, 669, 680, 680, 685, 687, 690, 711, 715, 719, 744, 748, 749, 751, 761, 764, 765, 772, 816, 848, 861, 873, 875, 906, 923, 951, 961, 965, 980, 983, 994, 1004, 1012, 1013, 1016, 1035, 1062, 1068, 1069, 1071, 1082, 1103, 1132, 1135, 1159, 1184, 1186, 1188, 1191, 1197, 1198, 1199, 1207, 1209, 1223, 1232, 1241, 1248, 1263, 1266, 1268, 1270, 1279, 1289, 1291, 1295, 1301, 1305, 1319, 1321, 1335, 1337, 1344, 1364, 1370, 1370, 1388, 1394, 1413, 1443, 1446, 1451, 1491, 1498, 1506, 1575, 1582, 1588, 1593, 1618, 1627, 1645, 1646, 1652, 1658, 1667, 1675, 1693, 1699, 1701, 1735, 1745, 1779, 1785, 1794, 1799, 1803, 1808, 1819, 1827, 1866, 1888, 1891, 1892, 1904

pag 162 - 1. Escreva um programa que leia do teclado dois conjuntos de nomes de fru-tas. Em seguida, apresente na tela a união (operador | ), a interseção (&) e a diferença simétrica (^). Utilizando a função len, verifique qual dos conjuntos tem mais elementos e apresente a diferença entre o maior e o menor (–).

In [ ]:
Cj1 = set()
Cj2 = set()
while True:
  Fruta = input('Digite a fruta para o conjunto 1 ou 0 para sair: ')
  if Fruta == '0':
    break
  else:
    Cj1.add(Fruta)
while True:
  Fruta = input('Digite a fruta para o conjunto 2 ou 0 para sair: ')
  if Fruta == '0':
    break
  else:
    Cj2.add(Fruta)
print(f'A união dos conjuntos é {Cj1 | Cj2}')
print(f'A interseção dos conjuntos é {Cj1 & Cj2}')
print(f'A diferença simétrica dos conjuntos é {Cj1 ^ Cj2}')
if len(Cj1) > len(Cj2):
  print(f'A diferença entre o maior e o menor conjunto é {len(Cj1) - len(Cj2)}')
  print(f'O maior conjunto é {Cj1}')
  print(f'O menor conjunto é {Cj2}')
else:
  print(f'A diferença entre o maior e o menor conjunto é {len(Cj2) - len(Cj1)}')
  print(f'O maior conjunto é {Cj2}')
  print(f'O menor conjunto é {Cj1}')





Digite a fruta para o conjunto 1 ou 0 para sair: uva
Digite a fruta para o conjunto 1 ou 0 para sair: manga
Digite a fruta para o conjunto 1 ou 0 para sair: laranja
Digite a fruta para o conjunto 1 ou 0 para sair: 0
Digite a fruta para o conjunto 2 ou 0 para sair: tomate
Digite a fruta para o conjunto 2 ou 0 para sair: abacate
Digite a fruta para o conjunto 2 ou 0 para sair: banana
Digite a fruta para o conjunto 2 ou 0 para sair: uva
Digite a fruta para o conjunto 2 ou 0 para sair: manga
Digite a fruta para o conjunto 2 ou 0 para sair: melão
Digite a fruta para o conjunto 2 ou 0 para sair: 0
A união dos conjuntos é {'laranja', 'melão', 'banana', 'uva', 'tomate', 'abacate', 'manga'}
A interseção dos conjuntos é {'uva', 'manga'}
A diferença simétrica dos conjuntos é {'laranja', 'melão', 'banana', 'abacate', 'tomate'}
A diferença entre o maior e o menor conjunto é 3
O maior conjunto é {'melão', 'banana', 'uva', 'tomate', 'abacate', 'manga'}
O menor conjunto é {'laranja', 'uva', 'manga'}


pag 162 2. Altere o programa do Exercício resolvido 6.2 da seguinte maneira: leia um número inteiro N (N < 30) e faça que o conjunto A tenha tamanho N e o con-junto B tenha tamanho 30-N.

In [ ]:
from random import randint
A = set()
B = set()
N = int(input('Digite um número inteiro menor que 30: '))
while len(A) < N:
  A.add(randint(1, 100))
while len(B) < 30 - N:
  B.add(randint(1, 100))

print(f'Conjunto A: {A}')
print(f'Conjunto B: {B}')


Digite um número inteiro menor que 30: 9
Conjunto A: {33, 97, 4, 12, 76, 79, 85, 53, 56}
Conjunto B: {3, 14, 15, 17, 20, 36, 44, 45, 51, 54, 63, 66, 70, 75, 77, 79, 81, 84, 85, 95, 100}


163 - 3. Escreva um programa que permaneça em laço lendo números inteiros do teclado. Esse laço termina quando for digitado zero ou qualquer valor negativo. O programa deve contar quantas vezes cada valor positivo foi digi-tado. Ao término do laço de leitura o programa deve mostrar quais valores foram digitados e quantas vezes cada um. Use um dicionário para resolver esse problema.


In [ ]:
Contagem = {}
try:
  while True:
    N = int(input('Digite um número inteiro ou menor igual a 0 para sair: '))
    if N <= 0:
      break
    else:
      Contagem[N] = Contagem.get(N, 0) + 1
except:
  print('Você deve digitar um número inteiro')

print(Contagem)

Digite um número inteiro ou menor igual a 0 para sair: 1
Digite um número inteiro ou menor igual a 0 para sair: 2
Digite um número inteiro ou menor igual a 0 para sair: 2
Digite um número inteiro ou menor igual a 0 para sair: 2
Digite um número inteiro ou menor igual a 0 para sair: 5
Digite um número inteiro ou menor igual a 0 para sair: 5
Digite um número inteiro ou menor igual a 0 para sair: 8
Digite um número inteiro ou menor igual a 0 para sair: 8
Digite um número inteiro ou menor igual a 0 para sair: 8
Digite um número inteiro ou menor igual a 0 para sair: 8
Digite um número inteiro ou menor igual a 0 para sair: 8
Digite um número inteiro ou menor igual a 0 para sair: 0
{1: 1, 2: 3, 5: 2, 8: 5}


163 - 4. Considere o seguinte conjunto de dados: Nome + (N1, N2, N3, N4). Nome re-presenta o nome de um aluno e deve ser usado como chave. N1, N2, N3, N4 representam as notas de provas desse aluno. Escreva um programa que leia os dados de Q alunos e apresente na tela se foram aprovados ou reprovados. O critério que garante a aprovação é que a média aritmética das 4 notas seja maior ou igual 6,0. Q é a quantidade de alunos, e esse valor deve ser lido do teclado no começo do programa. As notas devem ser exibidas com uma casa decimal.


In [ ]:
Cjdados = {}
try:
  Q = int(input('Digite a quantidade de alunos: '))
except:
  print('Você deve digitar um número inteiro')
while True:
  if Q == len(Cjdados):
    break
  else:
    Nome = input('Digite o nome do aluno: ')
    N1 = float(input('Digite a nota 1: '))
    N2 = float(input('Digite a nota 2: '))
    N3 = float(input('Digite a nota 3: '))
    N4 = float(input('Digite a nota 4: '))
    Cjdados[Nome] = (N1, N2, N3, N4)

for Nome, Notas in Cjdados.items():
  Media = sum(Notas) / len(Notas)
  if Media >= 6:
    print(f'{Nome} foi aprovado com média {Media:.1f}')
  else:
    print(f'{Nome} foi reprovado com média {Media:.1f}')


Digite a quantidade de alunos: 2
Digite o nome do aluno: Cão
Digite a nota 1: 4
Digite a nota 2: 3
Digite a nota 3: 5
Digite a nota 4: 6
Digite o nome do aluno: Gato
Digite a nota 1: 7
Digite a nota 2: 7
Digite a nota 3: 7
Digite a nota 4: 9
Cão foi reprovado com média 4.5
Gato foi aprovado com média 7.5


163 - 5. Refaça o exercício 4 alterando o critério de aprovação para o seguinte: das quatro notas, despreze a menor e calcule a média aritmética das outras três. Será considerado aprovado o aluno que tiver essa média maior ou igual a 6,0.

In [ ]:
Cjdados = {}
try:
  Q = int(input('Digite a quantidade de alunos: '))
except:
  print('Você deve digitar um número inteiro')
while True:
  if Q == len(Cjdados):
    break
  else:
    Nome = input('Digite o nome do aluno: ')
    N1 = float(input('Digite a nota 1: '))
    N2 = float(input('Digite a nota 2: '))
    N3 = float(input('Digite a nota 3: '))
    N4 = float(input('Digite a nota 4: '))
    Cjdados[Nome] = (N1, N2, N3, N4)

for Nome, Notas in Cjdados.items():
  Media = sum(Notas) / len(Notas)
  if Media >= 6:
    print(f'{Nome} foi aprovado com média {Media:.1f}')
  else:
    print(f'{Nome} foi reprovado com média {Media:.1f}')


Digite a quantidade de alunos: 2
Digite o nome do aluno: gato
Digite a nota 1: 2
Digite a nota 2: 7
Digite a nota 3: 9
Digite a nota 4: 7
Digite o nome do aluno: cão
Digite a nota 1: 5
Digite a nota 2: 3
Digite a nota 3: 2
Digite a nota 4: 4
gato foi aprovado com média 6.2
cão foi reprovado com média 3.5


163 - 6. Leia e armazene em um dicionário o nome, a idade e o número do telefone de seus contatos, sendo que a chave deve ser o nome. Ao digitar um string vazio para o nome, o programa interrompe a leitura. Apresente na tela os dados lidos em ordem alfabética pelo nome dos contatos. Em seguida, armazene os contatos em dois dicionários, utilizando como critério a idade: menores de 18 anos em um e os maiores em outro dicionário, eliminando o original. Apresente na tela os dois dicionários resultantes da separação.

In [ ]:
Arquivo = {}

try:
  while True:
    Nome = input('Digite o nome do contato ou espaço para sair: ')
    if Nome == ' ':
      break
    else:
      idade = int(input('Digite a idade do contato: '))
      Telefone = int(input('Digite o número do telefone do contato: '))
      Arquivo[Nome] = (idade, Telefone)
  Arquivo_novo = sorted(Arquivo.items())
  Arquivo_menores = {}
  Arquivo_maiores = {}
  for nome, numeros in Arquivo_novo:
    print(f'Nome: {nome} idade {numeros[0]} contato {numeros[1]}')
    if numeros[0] < 18:
      Arquivo_menores[nome] = numeros
    else:
      Arquivo_maiores[nome] = numeros
  print('Contatos menores de 18 anos')
  for nome, numeros in Arquivo_menores.items():
    print(f'Nome: {nome} idade {numeros[0]} contato {numeros[1]}')
  print('Contatos maiores de 18 anos')
  for nome, numeros in Arquivo_maiores.items():
    print(f'Nome: {nome} idade {numeros[0]} contato {numeros[1]}')
except:
 print('Você deve digitar um número inteiro para idade e contato')

Digite o nome do contato ou espaço para sair: rato
Digite a idade do contato: 22
Digite o número do telefone do contato: 22222222
Digite o nome do contato ou espaço para sair: boi
Digite a idade do contato: 6
Digite o número do telefone do contato: 66666666
Digite o nome do contato ou espaço para sair: pato
Digite a idade do contato: 44
Digite o número do telefone do contato: 44444444
Digite o nome do contato ou espaço para sair:  
Nome: boi idade 6 contato 66666666
Nome: pato idade 44 contato 44444444
Nome: rato idade 22 contato 22222222
Contatos menores de 18 anos
Nome: boi idade 6 contato 66666666
Contatos maiores de 18 anos
Nome: pato idade 44 contato 44444444
Nome: rato idade 22 contato 22222222


In [ ]:


a = 'teste'
b = 'casaco'
arq = open('teste.txt', 'w')
arq.write(a)
arq.close()
arq = open('teste.txt', 'a+')
arq.write(b)
arq.close()

print('Leitura')
arq = open('teste.txt', 'r')
a = arq.read()
arq.close()
print(a)

for s in open('teste.txt'):
  print(s)

Leitura
testecasaco
testecasaco


pag 182 - 1. Escreva um programa que leia um número inteiro N e grave em um arquivo em disco todos os números primos existentes no intervalo fechado [2, N], um número em cada linha. Sugestão: use uma função para determinar se o número é primo (veja o Exercício resolvido 5.1)

In [ ]:
def primos(N):
  L = []
  for i in range(2, N + 1):
    cont = 0
    for j in range(1, i + 1):
      if i % j == 0:
        cont += 1
    if cont == 2:
      L.append(i)
  return L

try:
  N = int(input('Digite um número inteiro: '))
except:
  print('Você deve digitar um número inteiro')
Lista_Primos = primos(N)
arq = open('primos.txt', 'w')
for i in Lista_Primos:
  arq.write(f'{str(i)}\n')

arq.close()
for s in open('primos.txt'):
  print(s)




Digite um número inteiro: 11
2

3

5

7

11



pag 182 2. Escreva um programa que grave o arquivo NUMEROS.TXT com 2000 números, um em cada linha, gerados com a função randint no intervalo fechado [1, 100.000].

In [ ]:
from random import randint
arq = open('numeros.txt', 'w')
for i in range(2000):
  arq.write(f'{str(randint(1,100000))}\n')
arq.close()

for s in open('numeros.txt'):
  print(s)

30420

65419

88466

80849

44605

43539

37237

11405

34862

92045

52502

74779

94182

85722

10541

4298

11409

1726

47794

17885

38763

87625

20895

45220

7942

87959

82717

59592

12011

51598

38528

50576

67207

39915

10219

25808

11811

3710

87803

12037

22066

73739

41720

5586

59345

64085

51536

8581

81589

62046

41555

40311

93233

88975

35483

19377

76867

44034

1989

34320

54609

38129

96569

17111

48439

59686

72434

99086

29701

55838

4571

22477

23245

40533

21649

82883

68863

80961

16708

73170

10892

81417

56497

98673

75505

77819

39201

40001

11623

67398

4393

47058

12786

37853

64730

29155

23968

79546

41589

74685

72563

76309

40582

70719

41331

14101

90103

79757

40628

57120

83441

99390

3842

67699

87131

78720

58954

84880

10651

28195

29935

84

17722

28599

8828

8519

64154

197

56137

14675

4649

29838

65879

76276

4388

61309

65480

34037

91823

75271

26627

29930

32194

52841

18162

4105

pag 183 3. Escreva um programa que leia o arquivo NUMEROS.TXT gerado no Exercí-cio 2, colocando-os em uma lista. Ordene a lista utilizando o método Bubble Sort e grave os números ordenados no arquivo “ORDENADOS.TXT”. Não use funções prontas de ordenação e recorra ao Exercício resolvido 4.11 para se lembrar do algoritmo Bubble Sort.


In [ ]:
def bublesort(L):
  for i in range(len(L)):
    for j in range(len(L)-1):
      if L[j] > L[j+1]:
        L[j], L[j+1] = L[j+1], L[j]
  return L

lista = []
arq = open('numeros.txt', 'r')
for s in arq:
  lista.append(int(s))
arq.close()

Lista_ordenada = bublesort(lista)

arq = open('ordenados.txt', 'w')
for i in Lista_ordenada:
  arq.write(f'{str(i)}\n')
arq.close()

for s in open('ordenados.txt'):
  print(s)

84

109

152

197

279

282

403

451

465

473

606

696

701

711

712

766

797

827

876

934

935

1223

1239

1266

1310

1333

1344

1364

1377

1377

1391

1450

1455

1465

1484

1519

1544

1566

1598

1645

1720

1726

1763

1771

1886

1937

1955

1970

1989

2125

2130

2160

2178

2212

2237

2241

2258

2272

2349

2381

2634

2722

2775

2799

2834

2861

2896

2898

2915

3059

3093

3094

3432

3468

3562

3579

3668

3689

3691

3710

3790

3796

3842

3889

3899

3928

3967

4004

4043

4084

4116

4118

4188

4237

4298

4355

4362

4388

4393

4412

4423

4447

4493

4503

4533

4543

4571

4633

4641

4649

4778

4978

4995

5215

5341

5356

5361

5401

5508

5510

5529

5578

5586

5658

5697

5697

5740

5799

5801

5952

5961

5972

6009

6049

6075

6147

6152

6257

6338

6340

6381

6407

6410

6438

6443

6464

6495

6496

6525

6600

6607

6732

6784

6795

6859

6866

6876

7004

7085

7122

7123

7136

7239

7263

7304

7310

7315

7419

7484

7513

75

pag 183 4. Repita o exercício anterior trocando o método de ordenação, para o método recursivo Quicksort (faça uma pesquisa sobre esse algoritmo). O Quicksort é um algoritmo consideravelmente mais veloz, na maioria dos casos, que o Bubble Sort. Use os programas 3 e 4 para constatar esse fato.


In [ ]:
def quicksort(arr):
    if len(arr) <= 1:
        return arr
    else:
        pivot = arr[0]
        less = [x for x in arr[1:] if x <= pivot]
        greater = [x for x in arr[1:] if x > pivot]
        return quicksort(less) + [pivot] + quicksort(greater)

lista = []
arq = open('numeros.txt' , 'r')
for s in arq:
  lista.append(s)
arq.close()

lista_ordenada = quicksort(lista)

arq = open('numeros.txt', 'w')
for i in lista_ordenada:
  arq.write(f'{str(i)}\n')
arq.close()

for s in open('numeros.txt'):
  print(s)


A saída de streaming foi truncada nas últimas 5000 linhas.
4488



44887



4490



44924



45058



45160



45236



45482



45548



45555



45594



45688



45690



45747



45765



45768



45791



45824



45870



45898



45908



45910



45915



45969



46011



46052



46068



46094



46128



46172



46173



46202



46273



4629



46357



46385



46402



46422



46423



46435



46436



46442



46458



46468



46477



4648



46567



46588



46598



46601



46634



46688



46728



46760



46780



46985



4707



47101



47114



4712



47157



47172



47183



47202



47225



47269



4727



47335



47359



47362



47417



47419



47433



47446



47461



47469



47625



47639



47698



47705



47714



47725



47777



47882



47883



4791



48003



48068



48105



48196



48223



4839



48390



48437



48446



48462



485



48507



48528



48551



48573



48681



48688



48699



488



48828





PAG 183 5. Escreva um programa que leia um arquivo texto de entrada contendo os sa-lários brutos dos funcionários de uma empresa e grave um arquivo de saída com os cálculos solicitados e descritos a seguir. O arquivo de entrada deve ter o nome “salario.txt” e conterá um valor em cada linha, sendo que, na primeira linha, estará escrito o título “Bruto”. Para cada salário devem ser calculados os descontos de INSS (previdência social), IR (imposto de renda) e salário líquido.

In [ ]:
arq = open('salario.txt', 'w')
arq.write('Bruto\n')
while True:
  salario = input('Digite o salário ou espaço para sair: ')
  if salario == ' ':
    break
  else:
    arq.write(f'{salario}\n')
arq.close()

arq = open('calculos.txt', 'w')

arq.write(f'SalBruto;AliqINSS;VALINSS;AliqIR;DeduçãoIR;VALIR;SalLiquido\n')

for i in open('salario.txt'):
  if i == 'Bruto\n':
    continue
  elif float(i) <= 1556.94:
    aliquota = 8
    inss = float(i) * 0.08
  elif float(i) >= 1556.95 and float(i) <= 2594.92:
    aliquota = 9
    inss = float(i) * 0.09
  elif float(i) >= 2594.93 and float(i) <= 5189.82:
    aliquota = 11
    inss = float(i) * 0.11
  else:
    aliquota = 'TETO'
    inss = 570.88

  if float(i) <= 1903.98:
    ir = 0
    deducao = 0
  elif float(i) >= 1903.99 and 2826.65:
    ir = 7.5
    deducao = 142.80
  elif float(i) >= 2826.66 and 3751.05:
    ir = 15
    deducao = 354.80
  elif float(i) >= 3751.06 and 4664.68:
    ir = 22.5
    deducao = 636.13
  else:
    ir = 27.5
    deducao = 869.36

  arq.write(f'{float(i)};{aliquota};{inss:.2f};{ir};{deducao};{ir/100*float(i):.2f},{float(i)-inss-(ir/100*float(i)):.2f}\n')

arq.close()

for i in open('calculos.txt'):
  print(i)

Digite o salário ou espaço para sair: 1228.90
Digite o salário ou espaço para sair: 2156.78
Digite o salário ou espaço para sair: 2298.37
Digite o salário ou espaço para sair: 3348.32
Digite o salário ou espaço para sair: 4573.12
Digite o salário ou espaço para sair: 4864.87
Digite o salário ou espaço para sair: 5031.87
Digite o salário ou espaço para sair: 8281.92 
Digite o salário ou espaço para sair:  
SalBruto;AliqINSS;VALINSS;AliqIR;DeduçãoIR;VALIR;SalLiquido

1228.9;8;98.31;0;0;0.00,1130.59

2156.78;9;194.11;7.5;142.8;161.76,1800.91

2298.37;9;206.85;7.5;142.8;172.38,1919.14

3348.32;11;368.32;7.5;142.8;251.12,2728.88

4573.12;11;503.04;7.5;142.8;342.98,3727.09

4864.87;11;535.14;7.5;142.8;364.87,3964.87

5031.87;11;553.51;7.5;142.8;377.39,4100.97

8281.92;TETO;570.88;7.5;142.8;621.14,7089.90



pag 184 6. Esse programa está baseado na leitura de um arquivo texto de entrada con-tendo números do registro de alunos. Esse arquivo deve ter o nome RA.TXT e conterá um RA em cada linha. Para cada número de RA presente no arquivo deve ser gerada uma senha, conforme as condições especificadas a seguir. Tanto o RA como a senha gerada devem ser gravados no arquivo de saída RASENHA.TXT, com o formato a seguir:
Condições para geração de senhas No início do programa, antes de efetuar a leitura do arquivo de entrada RA.TXT, o programa deve pedir que o usuário informe: ●
O tipo de senha: numérica (conterá apenas algarismos) ou alfanumérica (conterá letras maiúsculas e algarismos).
● O tamanho da senha: quantidade de caracteres que a mesma deve conter.



In [ ]:
import random
import string

arq = open('RA.txt', 'w')
while True:
  RA = input('Digite RA ou espaço para sair: ')
  if RA == ' ':
    break
  else:
    arq.write(f'{RA}\n')
arq.close()

def senha(tipo, tamanho):
  senha = ''
  if tipo == '1':
    for i in range(tamanho):
      senha += str(random.randint(0, 9))
  else:
    for i in range(tamanho):
      senha += random.choice(string.ascii_letters + string.digits)
  return senha

tipo = input('Digite 1 para senha numérica ou 2 para senha alfanumérica: ')
tamanho = int(input('Digite o tamanho da senha: '))

arq2 = open('RASENHA.txt', 'w')
arq2.write('RA;Senha\n')
for i in open('RA.txt'):
  arq2.write('i')
  arq2.write(f';{senha(tipo, tamanho)}\n')
arq.close()
arq2.close()

for i in open('RASENHA.txt'):
  print(i)


Digite RA ou espaço para sair: 123
Digite RA ou espaço para sair: 456
Digite RA ou espaço para sair:  
Digite 1 para senha numérica ou 2 para senha alfanumérica: 2
Digite o tamanho da senha: 7
RA;Senha

i;VN7pF88

i;C9bbCFz



pag 212 1. Escreva um programa que crie um banco de dados chamado “agenda.db”. Nesse BD deve existir uma tabela de contatos com os seguintes campos:
Nome do campo Tipo Descrição
NumContato integer (usar autonumeração) Chave primária.
Número inteiro de identificação gerado automaticamente no momento do cadastro.
Nome Text Nome da pessoa.
Cel Text Número do celular.
Tel Text Número do telefone fixo.
Email Text Endereço de e-mail.
Aniver Text Data de aniversário da pessoa
(armazenar como texto porque nem sempre se sabe o ano).






In [ ]:
import sqlite3
conexao = sqlite3.connect('agenda.db')
cursor = conexao.cursor()
sql = '''
  create table cadastro (NumContato integer primary key autoincrement,
  Nome text,
  Celular text,
  Telefone text,
  Email text,
  Aniversario text
  )'''
cursor.execute(sql)
conexao.commit()
conexao.close()


pag 213 2. Escreva um programa que exiba a agenda o exercício 1 e permita efetuar as seguintes ações:
cadastrar novas pessoas; alterar os telefones e o e-mail; excluir pessoas da agenda.


In [ ]:
import sqlite3
conexao = sqlite3.connect('agenda.db')
cursor = conexao.cursor()
while True:
  print('1 - Cadastrar nova pessoa')
  print('2 - Alterar telefone')
  print('3 - Alterar email')
  print('4 - Excluir pessoa')
  print('5 - Sair')

  opcao = input('Escolha uma opção: ')
  if opcao == '1':
    nome = input('Digite o nome: ')
    celular = input('Digite o número do celular: ')
    telefone = input('Digite o número do telefone: ')
    email = input('Digite o e-mail: ')
    aniversario = input('Digite a data de aniversário: ')
    Cad = (nome, celular, telefone, email, aniversario)
    sql = '''
    insert into cadastro (Nome, Celular, Telefone, Email, Aniversario)
    values (?, ?, ?, ?, ?)
    '''
    cursor.execute(sql, Cad)
    conexao.commit()
  elif opcao == '2':
    nome = input('Digite o nome: ')
    telefone = input('Digite o novo número do telefone: ')
    sql = '''
    update cadastro set Telefone = ? where Nome = ?
    '''
    cursor.execute(sql, (telefone, nome))
    conexao.commit()
  elif opcao == '3':
    nome = input('Digite o nome: ')
    email = input('Digite o novo e-mail: ')
    sql = '''
    update cadastro set Email = ? where Nome = ?
    '''
    cursor.execute(sql, (email, nome))
    conexao.commit()
  elif opcao == '4':
    nome = input('Digite o nome: ')
    sql = '''
    delete from cadastro where Nome = ?
    '''
    cursor.execute(sql, (nome,))
    conexao.commit()
  elif opcao == '5':
    break
  else:
    print('Opção inválida')
conexao.close()



1 - Cadastrar nova pessoa
2 - Alterar telefone
3 - Alterar email
4 - Excluir pessoa
5 - Sair
Escolha uma opção: 4
Digite o nome: Boi
1 - Cadastrar nova pessoa
2 - Alterar telefone
3 - Alterar email
4 - Excluir pessoa
5 - Sair
Escolha uma opção: 5


pag 213 3. Escreva um programa que crie um banco de dados chamado “musicas.db”. Esse BD deve conter três tabelas, mostradas a seguir.
Tabela “Musicas” contém a relação de músicas disponíveis no computador.

In [ ]:
import sqlite3
conexao = sqlite3.connect('musicas.db')
cursor = conexao.cursor()

sql = '''
  create table Musicas (nummusica integer primary key autoincrement,
  nomemus text, artista text, album text, ano integer, arquivo text)
  '''
cursor.execute(sql)
conexao.commit()

sql = '''
  create table Nomespl (nomepl text primary key, data date)
  '''
cursor.execute(sql)
conexao.commit()

sql = '''
  create table Playlist (nomepl Not NULL, nummusica Not NULL, primary key (nomepl, nummusica))
  '''
cursor.execute(sql)
conexao.commit()
conexao.close()

pag 214 4. Escreva um programa que permita exibir, inserir, alterar e excluir músicas no banco de dados do exercício 3.

In [ ]:
import sqlite3

conexao = sqlite3.connect('musicas.db')
cursor = conexao.cursor()

while True:
  print('1 - Exibir músicas')
  print('2 - Inserir música')
  print('3 - Alterar musica')
  print('4 - Excluir música')
  print('5 - Sair')
  try:
    opcao = int(input('Escolha uma opção: '))
  except:
    print('Opção inválida')
  if opcao == 1:
    sql = '''
    select * from Musicas
    '''
    cursor.execute(sql)
    for linha in cursor.fetchall():
      print(linha)
  elif opcao == 2:
    nomemus = input('Digite o nome da música: ')
    artista = input('Digite o nome do artista: ')
    album = input('Digite o nome do album: ')
    ano = input('Digite o ano de lançamento: ')
    arquivo = input('Digite o nome do arquivo: ')

    sql = '''
    insert into Musicas (nomemus, artista, album, ano, arquivo)
    values (?, ?, ?, ?, ?)
    '''
    cursor.execute(sql, (nomemus, artista, album, ano, arquivo))
    conexao.commit()
  elif opcao == 3:
    nomemus = input('Digite o nome da música: ')
    artista = input('Digite o nome do artista: ')
    album = input('Digite o nome do album: ')
    ano = input('Digite o ano de lançamento: ')
    arquivo = input('Digite o nome do arquivo: ')
    sql = '''
    update Musicas set nomemus = ?, artista = ?, album = ?, ano = ?, arquivo = ?
    where nomemus = ?
    '''
    cursor.execute(sql, (nomemus, artista, album, ano, arquivo, nomemus))
    conexao.commit()
  elif opcao == 4:
    nomemus = input('Digite o nome da música: ')
    sql = '''
    delete from Musicas where nomemus = ?
    '''
    cursor.execute(sql, (nomemus,))
    conexao.commit()
  elif opcao == 5:
    break
  else:
    print('Opção inválida')
conexao.close()

1 - Exibir músicas
2 - Inserir música
3 - Alterar musica
4 - Excluir música
5 - Sair
Escolha uma opção: 3
Digite o nome da música: boi
Digite o nome do artista: boi1
Digite o nome do album: boi1
Digite o ano de lançamento: 01.01.00
Digite o nome do arquivo: boi1
1 - Exibir músicas
2 - Inserir música
3 - Alterar musica
4 - Excluir música
5 - Sair
Escolha uma opção: 1
(1, 'boi', 'boi1', 'boi1', '01.01.00', 'boi1')
(2, 'vaca', 'vaca', 'vaca', '02.02.02', 'vaca')
1 - Exibir músicas
2 - Inserir música
3 - Alterar musica
4 - Excluir música
5 - Sair
Escolha uma opção: 4
Digite o nome da música: boi1
1 - Exibir músicas
2 - Inserir música
3 - Alterar musica
4 - Excluir música
5 - Sair
Escolha uma opção: 1
(1, 'boi', 'boi1', 'boi1', '01.01.00', 'boi1')
(2, 'vaca', 'vaca', 'vaca', '02.02.02', 'vaca')
1 - Exibir músicas
2 - Inserir música
3 - Alterar musica
4 - Excluir música
5 - Sair
Escolha uma opção: 4
Digite o nome da música: boi
1 - Exibir músicas
2 - Inserir música
3 - Alterar musica
4 - Exc

pag 214 5. Escreva um programa que permita exibir, inserir, alterar e excluir playlists (o nome e as músicas relacionadas) no banco de dados do exercício 3. Na inclusão de músicas, o programa só deve permitir inserir na tabela “Playlist” números de músicas que constarem da tabela “Musicas”.


In [ ]:
import sqlite3

conexao = sqlite3.connect('musicas.db')
cursor = conexao.cursor()

while True:
  print('1 - Exibir playList')
  print('2 - Inserir playList')
  print('3 - Alterar playList')
  print('4 - Excluir playList')
  print('5 - Sair')

  opcao = input('Escolha uma opção: ')
  if opcao == '1':
    sql = '''
    select * from Nomespl
    '''
    cursor.execute(sql)
    for linha in cursor.fetchall():
      print(linha)
  elif opcao == '2':
    nomepl = input('Digite o nome da playlist: ')
    data = input('Digite a data de criação da playlist: ')
    sql = '''
    insert into Nomespl (nomepl, data)
    values (?, ?)
    '''
    cursor.execute(sql, (nomepl, data))
    conexao.commit()
  elif opcao == '3':
    nomepl = input('Digite o nome da playlist a ser alterada')
    data = input('Digite a nova data de criação da playlist: ')

    sql = '''
    update Nomespl set data = ? where nomepl = ?
    '''
    cursor.execute(sql, (data, nomepl))
    conexao.commit()
    novonome = input('Deseja alterar o nome da playlist? (s/n)')
    if novonome == 's':
      nome = input('Digite o novo nome da playlist: ')
      sql = '''
      update Nomespl set nomepl = ? where nomepl = ?
      '''
      cursor.execute(sql, (nome, nomepl))
      conexao.commit()

  elif opcao == '4':
    nomepl = input('Digite o nome da playlist a ser excluida: ')
    sql = '''
    delete from Nomespl where nomepl = ?
    '''
    cursor.execute(sql, (nomepl,))
    conexao.commit()
  elif opcao == '5':
    break
  else:
    print('Opção inválida')



1 - Exibir playList
2 - Inserir playList
3 - Alterar playList
4 - Excluir playList
5 - Sair
Escolha uma opção: 2
Digite o nome da playlist: gato
Digite a data de criação da playlist: 07.07.07
1 - Exibir playList
2 - Inserir playList
3 - Alterar playList
4 - Excluir playList
5 - Sair
Escolha uma opção: 1
('vaca1', '01.01.09')
('gato', '07.07.07')
1 - Exibir playList
2 - Inserir playList
3 - Alterar playList
4 - Excluir playList
5 - Sair
Escolha uma opção: 4
Digite o nome da playlist a ser excluida: vaca1
1 - Exibir playList
2 - Inserir playList
3 - Alterar playList
4 - Excluir playList
5 - Sair
Escolha uma opção: 1
('gato', '07.07.07')
1 - Exibir playList
2 - Inserir playList
3 - Alterar playList
4 - Excluir playList
5 - Sair
Escolha uma opção: 5


pag 214 6. Escreva um programa que leia um arquivo texto do tipo CSV, com os dados separados pelo caractere ponto e vírgula “;” contendo
codigo;qtde;pccompra;pcvenda e importe esses dados na tabela vendas com a estrutura dada a seguir. Para testar seu programa, crie um arquivo de entrada com os dados anteriores,
utilizando um editor de textos comum. O banco de dados deve ter o nome “loja.db” e o nome da tabela deve ser “vendas”.

In [ ]:
import sqlite3

arq = open('cadastro.txt', 'w')

while True:
  codigo = input('Digite o código ou espaço para sair: ')
  qtde = input('Digite a quantidade ou espaço para sair: ')
  pccompra = input('Digite o preço de compra ou espaço para sair: ')
  pcvenda = input('Digite o preço de venda ou espaço para sair: ')
  arq.write(f'{codigo};{qtde};{pccompra};{pcvenda}\n')
  a = input('Deseja continuar? (s/n)')
  if a == 'n':
    break
arq.close()

conexao = sqlite3.connect('loja.db')
cursor = conexao.cursor()

sql = '''
  create table vendas (codigo integer primary key,
  qtde integer,
  pccompra real,
  pcvenda real
  )'''
cursor.execute(sql)
conexao.commit()

arq = open('cadastro.txt', 'r')
for i in arq:
  codigo, qtde, pccompra, pcvenda = i.split(';')
  sql = '''
  insert into vendas (codigo, qtde, pccompra, pcvenda)
  values (?, ?, ?, ?)
  '''
  cursor = conexao.cursor()
  cursor.execute(sql, (codigo, qtde, pccompra, pcvenda))
  conexao.commit()
arq.close()



Digite o código ou espaço para sair: 1
Digite a quantidade ou espaço para sair: 2
Digite o preço de compra ou espaço para sair: 3
Digite o preço de venda ou espaço para sair: 4
Deseja continuar? (s/n)s
Digite o código ou espaço para sair: 0
Digite a quantidade ou espaço para sair: 9
Digite o preço de compra ou espaço para sair: 8
Digite o preço de venda ou espaço para sair: 7
Deseja continuar? (s/n)n
